### PR TITLE
[android] Fixed crash.

### PR DESCRIPTION
**Pull request by yunikkk**

### CODE REVIEW

File: android/src/com/mapswithme/maps/location/LocationHelper.java [hunk 1]
```diff
@@ -209,6 +209,11 @@ public void run()
   {
     mLogger.d("ctor()");
 
+    // TODO consider refactoring.
+    // Actually we shouldn't initialize Framework here,
+    // to allow app components to retrieve location updates without all heavy framework's stuff initialized.
+    // For now this is necessary to connect mModeChangeListener below.
+    MwmApplication.get().initNativeCore();
```
Comment (ygorshenin) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: How this may happen, that LocationHelper is initialized but framework is not?

Comment (ygorshenin) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: And why not to just skip nativeSetListener if framework is not initialized?

Comment (trashkalmar) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: Framework is initialized (in general) much later than other components. And here `nativeSetListener()` is the crucial operation for this class.

Comment (ygorshenin) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: As far as I know framework is not needed when track recording is enabled and application is killed.

Comment (ygorshenin) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: I.e. when track recording works in background, in daemon mode, framework is not needed, right?

**Comment by mpimenov** [comment]
Merged in #4066.

File: android/src/com/mapswithme/maps/location/LocationHelper.java (hunk 1 above)
Comment (trashkalmar) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: It seems framework required anyway. At least to `nativeSetListener()`

Comment (yunikkk) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: Yep, now it's required at least to set that listener. Left a TODO to refactor that.

**Pull request closed without merging**